src/datasets/arrow_dataset.py

```diff
@@ -3464,6 +3511,11 @@ def init_buffer_and_writer():
                                 writer.write_row(example)
                             elif isinstance(example, pd.DataFrame):
                                 writer.write_row(pa.Table.from_pandas(example))
+                            elif config.POLARS_AVAILABLE:
+                                import polars as pl
+
+                                if isinstance(example, pl.DataFrame):
+                                    writer.write_table(example.to_arrow())
```

Comment:
Here you should write_row instead of write_table

Reply:
fixed